src/sparseml/exporters/transforms/kv_cache/cache_keys_and_values.py

```diff
@@ -255,18 +263,66 @@ def create_cache(
         cache_output_dims,
     )
 
+    graph = ONNXGraph(model)
+
+    if node.op_type == "QLinearMatMul" and cache_input_idx == 1:
+        cache_input_idx = 3  # QLinearMatMul B matrix is at idx 3, not 1
+
+    cache_parent = graph.get_node_single_parent(node, index=cache_input_idx)
+    if isinstance(cache_parent, NodeProto) and cache_parent.op_type == "Transpose":
```

Comment:
does this guarantee the cache input dims will be in the exact expected order of the pipeline/engine?

Reply:
i think so. where do you see any inconsistencies?